src/main/java/org/folio/fqm/repository/EntityTypeRepository.java

```diff
@@ -48,30 +53,45 @@ public class EntityTypeRepository {
   private final DSLContext jooqContext;
   private final ObjectMapper objectMapper;
 
+  @Value("${mod-fqm-manager.entity-type-cache-timeout-seconds:3600}")
+  private long cacheDurationSeconds;
+  private final Cache<String, Map<UUID, EntityType>> entityTypeCache = Caffeine.newBuilder().expireAfterWrite(cacheDurationSeconds, TimeUnit.SECONDS).build();
+
   public Optional<EntityType> getEntityTypeDefinition(UUID entityTypeId, String tenantId) {
     return getEntityTypeDefinitions(Collections.singleton(entityTypeId), tenantId).findFirst();
   }
 
   public Stream<EntityType> getEntityTypeDefinitions(Collection<UUID> entityTypeIds, String tenantId) {
```

Comment:
The major change in this method is that instead of only loading some ETs in some cases and all of them in others, it just always loads them all. Then, all entity type defs are just served up from the cache